### ISSUE TITLE
[FR] Weighted geometric standard deviation

### ISSUE THREAD

**Issue by maxfrei750** [scipy/scipy#13424]
## Pitch
Since the `stats` module offers functions to calculate the weighted geometric mean and the geometric standard deviation, it would IMHO make sense to provide a _weighted_ geometric standard deviation.

## Potential solutions
As basis for the implementation, the following sources could be used, if they are actually correct (naturally better sources would be good):

1. [R implementation](https://rdrr.io/cran/functClust/src/R/stats.R#sym-wgsd)

2. [Formula given by Prof. Guillermo Enrique Ramos](https://www.researchgate.net/post/How-can-I-calculate-the-value-of-the-geometric-standard-deviation-taking-into-account-weight/53721a9ed5a3f2d33c8b4607/citation/download)

A basic implementation could look like this (based on 2):
```
def gstd(a, weights=None):

    log_a = np.log(a)

    weighted_average = np.average(log_a, weights=weights)
    weighted_variance = np.average((log_a - average) ** 2, weights=weights)

    return np.exp(np.sqrt(weighted_variance))
```

## Problems
1. Apparently, the calculation and interpretation of the weighted geometric standard deviation is no trivial subject.
2. So far, I couldn't find reliable reference implementations or precalculated test values.

## Necessary steps to implement this feature
1. Discussion on how to implement it (help would be appreciated).
2. Somebody willing to implement it (that would be me).
3. A reference implementation or precalculated test values from literature.

**Comment by mdhaber** [comment]
Crossref gh-6931 for other relevant information, although `gstd` is not on that list.

**Comment by maxfrei750** [comment]
@mdhaber As far as I can see, the only thing missing for this feature is somebody competent and confident enough, to comment on the correct way to implement this. Do you know anybody that might be suitable and willing that you could ping, to take a brief look at the _Potential solutions_? Thanks!

**Comment by mdhaber** [comment]
Yes, one thing this needs is a reviewer. The other thing that this needs is interests from others in adding this to the API. The best way to proceed for both is to send an email to the [scipy-dev mailing list](https://svn.scipy.org/scipylib/mailing-lists.html). Thanks!

**Comment by maxfrei750** [comment]
Thanks for the hint. I'll try to find suitable reviewers and check if people are even interested in this. :smile:

**Comment by j-bowhay** [comment]
I removed the link to the R package as it is GPL licensed so the code cannot be looked at while creating an implementation for SciPy

**Comment by maxfrei750** [comment]
@j-bowhay Good catch. Sorry for nearly violating the license/infecting scipy.

**Comment by lutefiskhotdish** [comment]
1. Should there be a way to get sample standard deviation rather than population standard deviation?  For example, `np.std` has the [parameter `doff` which can be set to 1 to get sample standard deviation](https://stackoverflow.com/questions/34050491/standard-deviation-in-numpy#34050706).

2. Weighted standard geometric deviation is the de-logged weighted standard deviation of logged values.  This might suggest building the geometric version on top of a function that calculates weighted standard (arithmetic) deviation:

```
def weighted_std(a, weights=None):
    ...  # See core of the code in Potential Solutions above.

def gstd(a, weights=None):
    log_a = np.log(a)
    log_sd = weighted_std(log_a, weights=weights)
    return np.exp(log_sd)
```

**Comment by lorentzenchr** [comment]
Wouldn’t introducing weighted variance be the first central step? Maybe there is already an issue, maybe even in numpy?
